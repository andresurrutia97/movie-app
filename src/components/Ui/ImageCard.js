import React from 'react';
import PropTypes from 'prop-types';

const ImageCard = ({ imgUrl }) => (
  <div
    className="custom-poster-card rounded-lg shadow-lg"
    style={{ backgroundImage: `url('${imgUrl}')` }}
  />
);

ImageCard.propTypes = {
  imgUrl: PropTypes.string.isRequired,
};

export default ImageCard;
