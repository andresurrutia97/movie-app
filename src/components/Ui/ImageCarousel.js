import React from 'react';
import PropTypes from 'prop-types';

const ImageCarousel = ({ title, children }) => {
  return (
    <div className="my-4">
      <div className="text-xl text-white">{title}</div>
      <div className="flex overflow-x-auto custom-scrollbar overflow-y-hidden">
        <div className="flex py-5 ">{children}</div>
      </div>
    </div>
  );
};

ImageCarousel.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string,
};

ImageCarousel.defaultProps = {
  children: null,
  title: '',
};

export default ImageCarousel;
