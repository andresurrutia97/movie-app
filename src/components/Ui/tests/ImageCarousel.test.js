import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ImageCarousel from '../ImageCarousel';
import ImageCard from '../ImageCard';

describe('<ImageCarousel/>', () => {
  const component = <ImageCarousel />;

  it('should render correcty ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain an instance of an <ImageCard/> ', () => {
    const wrapper = shallow(
      <ImageCarousel>
        <ImageCard imgUrl="path" />
      </ImageCarousel>
    );
    expect(wrapper.find(ImageCard)).toHaveLength(1);
  });
});
