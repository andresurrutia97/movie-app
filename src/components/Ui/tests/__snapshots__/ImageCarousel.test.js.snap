// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageCarousel/> should render correcty  1`] = `
<div
  className="my-4"
>
  <div
    className="text-xl text-white"
  >
    
  </div>
  <div
    className="flex overflow-x-auto custom-scrollbar overflow-y-hidden"
  >
    <div
      className="flex py-5 "
    />
  </div>
</div>
`;
